subrootpaths.go

```diff
@@ -133,8 +127,15 @@ func GetSubrootPaths(squareSize uint, idxStart uint, shareCount uint) ([][][]int
 		return nil, srpInvalidShareCount
 	}
 
+	isn := idxStart + shareCount
+	if isn < idxStart || isn < shareCount {
+		return nil, errors.New("invalid idxStart+shareCount value")
```

Comment:
[optional] it may help future developers if this error string included additional debugging information

```suggestion
		return nil, fmt.Sprintf("invalid isn %v must be less than idxStart %v and shareCount %v", isn, idxStart, shareCount)
```

If this proposal isn't accepted, the existing static error string may be hoisted to line 13 of this file so that all error strings are defined in one location for this file.

Reply:
[blocking] is it possible to include a unit test that demonstrates this error is returned under the expected circumstances?

Reply:
I am purposefully not sending in a PR given it can be exploited, I had spoken to y'all offline for responsible disclosure of the bug :-)

Reply:
I've hoisted the error out as you requested @rootulp

Reply:
> I am purposefully not sending in a PR given it can be exploited

I don't think we should be concerned about adding a unit test for this error. A unit test gives maintainers greater confidence that the bug has been addressed.

If an attacker wants to exploit this integer overflow, they would be able to do so irrespective of this unit test.